clientv3/watch.go

```diff
@@ -61,6 +61,11 @@ func (wr *WatchResponse) Err() error {
 	return nil
 }
 
+// IsProgressNotify returns true if the WatchResponse is progress notification.
+func (wr *WatchResponse) IsProgressNotify() bool {
+	return len(wr.Events) == 0 && !wr.Canceled
```

Comment:
This is `Canceled` from `WatchResponse`. When we compact beyond watch revision, the watcher gets canceled, and the `WatchResponse.Canceled` is set `true`.

@hongchaodeng Do you mean close the channel in clientv3? We close channel when
`wr.Canceled` is true, which means it's not progress notify.